Mathlib/GroupTheory/SpecificGroups/Alternating/KleinFour.lean

```diff
@@ -0,0 +1,240 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.GroupTheory.SpecificGroups.Alternating.Centralizer
+import Mathlib.GroupTheory.SpecificGroups.KleinFour
+import Mathlib.GroupTheory.Sylow
+
+/-! # The Klein Four subgroup of an alternating group on 4 letters
+
+Let `α` be a finite type.
+
+* `alternatingGroup.center_eq_bot`: when `4 ≤ Nat.card α`,
+then center of `alternatingGroup α` is trivial.
+
+We now assume that `Nat.card α = 4`.
+
+* `alternatingGroup.kleinFour` : the subgroup
+of `alternatingGroup α` generated by permutations of cycle type (2, 2).
+
+* `alternatingGroup.kleinFour_isKleinFour`:
+`alternatingGroup.kleinFour α` satisfies `IsKleinFour`.
+(When `4 < Nat.card α`, it is equal to `⊤`, when `Nat.card α < 4`, it is trivial.)
+
+* `alternatingGroup.two_sylow_eq_kleinFour_of_card_eq_four`:
+All `2`-sylow subgroups of `alternatingGroup α` are equal
+to `kleinFour α`.
+
+* `alternatingGroup.characteristic_kleinFour`:
+the subgroup `alternatingGroup.kleinFour α` is characteristic.
+
+* `alternatingGroup.normal_kleinFour`:
+the subgroup `alternatingGroup.kleinFour α` is normal.
```

Comment:
Probably not worth including in the documentation since it follows immediately from the previous.